modules/urlsigning-service-impl/src/main/java/org/opencastproject/security/urlsigning/provider/impl/AbstractUrlSigningProvider.java

```diff
@@ -130,62 +146,69 @@ public void updated(Dictionary<String, ?> properties) throws ConfigurationExcept
     }
 
     // Collect configuration in a new map so we don't partially override the old one in case of error
-    Map<String, KeyEntry> keys = new HashMap<>();
+    TreeMap<String, Key> urls = new TreeMap<>();
+
+    // Temporary list of key entries to simplify building up the keys
+    Map<String, Key> keys = new HashMap<>();
 
     Enumeration<String> propertyKeys = properties.keys();
     while (propertyKeys.hasMoreElements()) {
       String propertyKey = propertyKeys.nextElement();
 
-      String keyEntryProperty = StringUtils.removeStart(propertyKey, KEY_ENTRY_PREFIX + ".");
-      if (keyEntryProperty == propertyKey) continue;
+      if (!propertyKey.startsWith(KEY_PROPERTY_PREFIX + ".")) {
+        /* This is not a key configuration property -> ignore */
+        continue;
+      }
 
-      String[] parts = Arrays.stream(keyEntryProperty.split("\\."))
+      String[] parts = Arrays.stream(propertyKey.split("\\."))
               .map(String::trim)
               .toArray(String[]::new);
-      if (parts.length != 2) {
-        throw new ConfigurationException(propertyKey, "wrong property key format");
+      if (parts.length != 3) {
+        /* We expected [KEY_PROPERTY_PREFIX, id, attribute here */
+        throw new ConfigurationException(propertyKey, "Wrong property key format");
       }
 
-      String id = parts[0];
-      KeyEntry currentKeyEntry = keys.computeIfAbsent(id, __ -> new KeyEntry());
+      String id = parts[1];
+      Key currentKey = keys.computeIfAbsent(id, __ -> new Key(id));
 
-      String attribute = parts[1];
+      String attribute = parts[2];
       String propertyValue = StringUtils.trimToNull(Objects.toString(properties.get(propertyKey), null));
       if (propertyValue == null) {
-        throw new ConfigurationException(propertyKey, "can't be null or empty");
+        throw new ConfigurationException(propertyKey, "Can't be null or empty");
       }
       switch (attribute) {
         case ORGANIZATION:
-          currentKeyEntry.organization = propertyValue;
+          currentKey.organizationId = propertyValue;
           break;
         case URL:
-          if (keys.values().stream()
-                  .map(keyEntry -> keyEntry.url)
-                  .filter(Objects::nonNull)
-                  .anyMatch(url -> propertyValue.startsWith(url) || (url != null && url.startsWith(propertyValue)))) {
+          if (urls.keySet().stream().anyMatch(v -> propertyValue.startsWith(v) || (v.startsWith(propertyValue)))) {
             throw new ConfigurationException(propertyKey,
-                    "there is already a key configuration for a URL with the prefix " + propertyValue);
+                    "There is already a key configuration for a URL with the prefix " + propertyValue);
           }
-          currentKeyEntry.url = propertyValue;
+          /* We explicitely support multiple URLs that map to the same key */
```

Comment:
As properties is a set, it is currently not possible to add multiple URLs. That will be added by https://opencast.jira.com/browse/MH-13513 later.